probe/elasticsearch.go

```diff
@@ -0,0 +1,562 @@
+						common.ElasticNodeAvailabilityGauge.WithLabelValues(esNode.Cluster, esNode.Name).Set(0)
+						common.ClusterErrorsCount.WithLabelValues(es.clusterName).Add(1)
+						log.Error(err)
+					}
+				}(node)
+			}
+			sem.Wait()
+		}
+	}
+}
+
+func probeElasticsearchNode(node *common.Node, timeout time.Duration, username, password string) error {
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+	client := &http.Client{
+		Timeout: timeout,
+	}
+
+	probingURL := fmt.Sprintf("%v://%v:%v/_cat/health?v", node.Scheme, node.Ip, node.Port)
+	log.Debug("Start probing ", node.Name)
+
+	start := time.Now()
+	req, err := http.NewRequest("GET", probingURL, nil)
+	if err != nil {
+		return err
+	}
+	req.SetBasicAuth(username, password)
+	resp, err := client.Do(req)
+	if err != nil {
+		log.Debug("Probing failed for ", node.Name, ": ", probingURL, " ", err.Error())
+		return err
+	}
+	durationMilliSec := float64(time.Since(start).Milliseconds())
+
+	log.Debug("Probe result for ", node.Name, ": ", resp.Status)
+	if resp.StatusCode != 200 {
+		log.Error("Probing failed for ", node.Name, ": ", probingURL, " ", resp.Status)
+		return fmt.Errorf("ES Probing failed")
+	}
+
+	common.ElasticNodeAvailabilityGauge.WithLabelValues(node.Cluster, node.Name).Set(1)
+	common.NodeCatLatencySummary.WithLabelValues(node.Cluster, node.Name).Observe(durationMilliSec)
+
+	return nil
+}
+
+func initEsClient(scheme, endpoint, username, passsword string) (*elasticsearch7.Client, error) {
+	http.DefaultTransport.(*http.Transport).TLSClientConfig = &tls.Config{InsecureSkipVerify: true}
+
+	cfg := elasticsearch7.Config{
+		Addresses: []string{
+			fmt.Sprintf("%v://%v", scheme, endpoint),
+		},
+		Username: username,
+		Password: passsword,
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{
+				InsecureSkipVerify: true,
+			},
+		},
+	}
+	es, err := elasticsearch7.NewClient(cfg)
+	if err != nil {
+		log.Fatalf("Error creating the client: %s", err)
+		return nil, err
+	}
+	return es, nil
+}
+
+func (es *EsProbe) deleteDocument(index, documentID string) error {
+	start := time.Now()
+	res, err := es.client.Delete(
+		index,
+		documentID)
+	durationMilliSec := float64(time.Since(start).Milliseconds())
+
+	if err != nil {
+		return errors.Wrapf(err, "Failed to delete document %s on %s:%s", documentID, es.clusterName, index)
+	}
+	defer res.Body.Close()
+
+	if res.IsError() {
+		return errors.Errorf("Error delete document %s on %s:%s: %s", documentID, es.clusterName, index, res.String())
+	}
+
+	common.ClusterLatencySummary.WithLabelValues(es.clusterName, index, "delete").Observe(durationMilliSec)
+	common.ClusterLatencyHistogram.WithLabelValues(es.clusterName, index, "delete").Observe(durationMilliSec)
+
+	return nil
+}
+
+func (es *EsProbe) getDocument(index, documentID string) error {
+	start := time.Now()
+	res, err := es.client.Get(
```

Comment:
Can you check that the document is indeed downloaded and not waiting for you to call the reader

Reply:
I confirm the http call is immediately issue